wandb/sdk/artifacts/artifact.py

```diff
@@ -1647,12 +1647,16 @@ def download(
 
         Raises:
             ArtifactNotLoggedError: If the artifact is not logged.
+            ValueError: If the artifact is attempted to be downloaded in offline mode.
         """
         self._ensure_logged("download")
 
         root = FilePathStr(str(root or self._default_root()))
         self._add_download_root(root)
 
+        if env.is_offline():
+            raise ValueError("Cannot download artifacts in offline mode.")
```

Comment:
suggestion: Not sure if there's already a custom exception for environment errors (defer to SDK team if there is).  If not, `RuntimeError` seems much more accurate + appropriate here.
```suggestion
            raise RuntimeError("Cannot download artifacts in offline mode.")
```

Context: typically would expect ValueError instead if -- for example -- one of the arguments passed to this method had a valid type but invalid value (eg a malformatted string)